linera-views/src/backends/scylla_db.rs

```diff
@@ -412,46 +460,127 @@ impl ScyllaDbClient {
         Ok(rows.next().is_some())
     }
 
+    fn attempt_sticky_shard_policy_creation(
+        &self,
+        partition_key: &[u8],
+        token: Option<Token>,
+        cache: &mut LruCache<Vec<u8>, LoadBalancingPolicyCacheEntry>,
+    ) -> Arc<dyn LoadBalancingPolicy> {
+        match StickyShardPolicy::new(
+            &self.session,
+            &self.namespace,
+            partition_key,
+            token,
+            ScyllaDbClient::build_default_policy(),
+        ) {
+            Ok(policy) => {
+                let policy = Arc::new(policy);
+                cache.push(
+                    partition_key.to_vec(),
+                    LoadBalancingPolicyCacheEntry::Ready(policy.clone()),
+                );
+                policy
+            }
+            Err(error) => {
+                // Cache that the policy creation failed, so we don't try again too soon, and don't
+                // recalculate the token if not needed.
+                let token = match error {
+                    ScyllaDbStoreInternalError::MissingTokenEndpoints(token) => Some(token),
+                    _ => None,
+                };
+                cache.push(
+                    partition_key.to_vec(),
+                    LoadBalancingPolicyCacheEntry::NotReady(Timestamp::now(), token),
+                );
+                ScyllaDbClient::build_default_policy()
+            }
+        }
+    }
+
+    // Returns a batch query with a sticky shard policy, that always tries to route to the same
+    // ScyllaDB shard.
+    // Should be used only on batches where all statements are to the same partition key.
+    fn get_sticky_batch_query(
+        &self,
+        partition_key: &[u8],
+    ) -> Result<Batch, ScyllaDbStoreInternalError> {
+        // Since we assume this is all to the same partition key, we can use an unlogged batch.
+        // We could use a logged batch to get atomicity across different partitions, but that
+        // comes with a huge performance penalty (seems to double write latency).
+        let mut batch_query = Batch::new(BatchType::Unlogged);
+        // Getting the sticky shard policy does some serializing and hashing under the hood, so
+        // we cache the policy to avoid that extra work.
+        let policy = {
+            let mut cache = self
+                .batch_load_balancing_policies
+                .lock()
+                .map_err(|_| ScyllaDbStoreInternalError::PoisonedMutex)?;
+            if let Some(policy) = cache.get(partition_key) {
+                match policy {
+                    LoadBalancingPolicyCacheEntry::Ready(policy) => policy.clone(),
+                    LoadBalancingPolicyCacheEntry::NotReady(timestamp, token) => {
+                        if Timestamp::now().delta_since(*timestamp)
```

Comment:
Maybe we should use `Instance` here instead of `Timestamp`? Our `linera_base` timestamp type is just to define and serialize timestamps in the protocol as `u64`s. Since this here is only used locally I'd go with the standard library. Also, this expression here could then be `timestamp.elapsed()`.